lib/merkle-tree.js

```diff
@@ -221,24 +222,28 @@ class MerkleTreeBatch {
 
     if (this.upgraded) this._commitUpgrade()
 
+    const batch = this.tree.storage.createWriteBatch()
     for (let i = 0; i < this.nodes.length; i++) {
       const node = this.nodes[i]
-      this.tree.unflushed.set(node.index, node)
+      batch.addTreeNode(node)
     }
+
+    return batch.flush()
   }
 
   _commitUpgrade () {
     // TODO: If easy to detect, we should refuse an trunc+append here without a fork id
     // change. Will only happen on user error so mostly to prevent that.
 
+    const batch = this.tree.storage.createWriteBatch()
     if (this.ancestors < this.treeLength) {
       if (this.ancestors > 0) {
         const head = 2 * this.ancestors
         const ite = flat.iterator(head - 2)
 
         while (true) {
           if (ite.contains(head) && ite.index < head) {
-            this.tree.unflushed.set(ite.index, blankNode(ite.index))
+            batch.addTreeNode(blankNode(ite.index))
```

Comment:
now that we have rocks we can just delete nodes! Can you add that api for tree nodes to storage